.github/copilot-instructions.md

```diff
@@ -0,0 +1,314 @@
+- **System tests: 15-20 minutes** - Browser automation with Capybara/Selenium
+- **Asset builds: 2-5 minutes each** - CSS and JavaScript compilation
+
+### Authentication Requirements
+- **NPM Token required**: Private package access needs `NPM_TOKEN` environment variable
+- **AWS LocalStack**: Required for S3, DynamoDB, and other AWS service mocking
+- **Database credentials**: Must match config/database.yml settings
+
+### Version Dependencies (CRITICAL)
+- **Ruby 3.4.4**: Exact version required - Gemfile.lock specifies this
+- **Node.js 20+**: Required for esbuild and modern JS features
+- **MySQL 5.7**: Database schema depends on this version
+- **Bundler 2.6.9**: Locked version in Gemfile.lock
+
+### Known Limitations
+- Ruby version managers (rbenv/rvm) recommended for version management
+- Some JS packages require private NPM access (will fail in some environments)  
+- Docker services must be running before Rails server starts
+- Database must be properly configured with utf8mb4 collation
+- Large codebase - initial setup can take 30+ minutes total
+
+## Important Directories
+
+### Application Structure
+- `app/controllers/` - Rails controllers
+- `app/models/` - ActiveRecord models and business logic
+- `app/commands/` - Mandate command objects for business logic (see Command Pattern below)
+- `app/javascript/` - React/TypeScript frontend code
+- `app/css/` - PostCSS stylesheets with Tailwind
+- `app/views/` - HAML view templates
+- `test/` - All test files (unit, integration, system)
+- `test/system/` - Capybara system tests
+- `test/javascript/` - Jest test files
+
+### Command Pattern with Mandate
+The `/app/commands` directory contains command objects that encapsulate business logic using the Mandate gem. This pattern promotes:
+- Single Responsibility Principle
+- Testable business logic separation from controllers
+- Consistent error handling and callbacks
+
+#### Command Structure
+Commands follow a consistent pattern:
+
+```ruby
+class User::Update
+  include Mandate
+  include Mandate::Callbacks  # Optional, for success/failure callbacks
+
+  initialize_with :user, :params  # Defines constructor parameters
+
+  def call  # Single method that performs the operation
+    # Business logic here
+    abort!(errors) if errors.present?  # Stops execution and triggers failure callback
+  end
+
+  memoize  # Caches the result of expensive operations
+  def errors
+    # Validation logic
+  end
+
+  private
+  # Helper methods
+end
+```
+
+#### Calling Commands
+Commands are invoked using the `.()` syntax:
+
+```ruby
+# Simple call
+result = User::Update.(current_user, params)
+
+# With callbacks (common in API controllers)
+cmd = User::Update.(current_user, params)
+cmd.on_success { render json: {} }
+cmd.on_failure { |errors| render_400(:failed_validations, errors: errors) }
+```
+
+#### Key Patterns
+- **`initialize_with`**: Defines required and optional parameters for the command constructor
```

Comment:
Mention positional params (for everything required), named params (for optional),  and named with Mandate::NO_DEFAULT for rare situations (find why we use this)